docs/OpenShift GitOps Usage Guide.md

```diff
@@ -0,0 +1,1291 @@
+```
+
+
+Now we can edit the argocd cr and add the oidc configuration to enable our keycloak authentication. You can use 
+
+`$ kubectl edit argocd -n openshift-gitops`.
+
+Your argocd cr should look like this:
+
+```
+apiVersion: argoproj.io/v1alpha1
+kind: ArgoCD
+metadata:
+  creationTimestamp: null
+  name: argocd
+  namespace: argocd
+spec:
+  resourceExclusions: |
+    - apiGroups:
+      - tekton.dev
+      clusters:
+      - '*'
+      kinds:
+      - TaskRun
+      - PipelineRun
+  oidcConfig: |
+    name: OpenShift Single Sign-On
+    issuer: https://keycloak.example.com/auth/realms/myrelam
+    clientID: argocd
+    clientSecret: $oidc.keycloak.clientSecret
+    requestedScopes: ["openid", "profile", "email", "groups"]
+  server:
+    route:
+      enabled: true
+```
+
+**Make sure **that: - **issuer** ends with the correct realm (in this example *myrealm*) - **clientID** is set to the Client ID you configured in Keycloak - **clientSecret** points to the right key you created in the *argocd-secret* Secret - **requestedScopes** contains the *groups* claim if you didn't add it to the Default scope.
+
+### Login via Keycloak (RHSSO)
+
+![image alt text](assets/image_21.png)
+
+### **Keycloak Identity brokering with OpenShift OAuthClient**
+
+Prior to configuring OpenShift 4 Identity Provider, please locate the correct OpenShift 4 API URL up. The easiest way to obtain it is to invoke the following command (this might require installing jq command separately) 
+
+`curl -s -k -H "Authorization: Bearer $(oc whoami -t)" https://<openshift-user-facing-api-url>/apis/config.openshift.io/v1/infrastructures/cluster | jq ".status.apiServerURL". `
+
+In most cases, the address will be protected by HTTPS. Therefore, it is essential to configure X509_CA_BUNDLE in the container and set it to /var/run/secrets/kubernetes.io/serviceaccount/ca.crt. Otherwise, Keycloak won’t be able to communicate with the API Server.
+
+Go to Identity Providers and select Openshift v4. Set the Base Url to OpenShift 4 API URL. Client ID as keycloak-broker(can be anything) and Client Secret to any secret that you want to set in Step 7.
+
+## **Registering an OAuth Client** 
+
+```
+kind: OAuthClient
+apiVersion: oauth.openshift.io/v1
+metadata:
+ name: keycloak-broker 
+secret: "..." 
+redirectURIs:
+"https://keycloak-keycloak.apps.dev-svc-4.7-020201.devcluster.openshift.com/auth/realms/myrealm/broker/openshift-v4/endpoint" 
+grantMethod: prompt 
+```
+
+1. The name of the OAuth client is used as the client_id parameter when making requests to <namespace_route>/oauth/authorize and <namespace_route>/oauth/token.\
+2. The secret is used as the client_secret parameter when making requests to <namespace_route>/oauth/token.\
+3. The redirect_uri parameter specified in requests to <namespace_route>/oauth/authorize and <namespace_route>/oauth/token must be equal to or prefixed by one of the URIs listed in the redirectURIs parameter value. Refer to the screenshot in Step 6. \
```

Comment:
remove / at the end, also what is the screenshot in Step 6? I don't see a step 6 here